integration-webcert/src/main/java/se/inera/intyg/minaintyg/integration/webcert/converter/data/value/IcfValueConverter.java

```diff
@@ -17,14 +18,23 @@ public CertificateDataValueType getType() {
 
   @Override
   public CertificateQuestionValue convertToValue(CertificateDataElement element) {
-    final var value = ((CertificateDataIcfValue) element.getValue()).getText();
-    // TODO: Handle icf-codes
-    return getText(value);
+    final var icfCodes = ((CertificateDataIcfValue) element.getValue()).getIcfCodes();
+    final var text = ((CertificateDataIcfValue) element.getValue()).getText();
+
+    if (icfCodes != null) {
+      final var collectionsLabel = ((CertificateDataConfigIcf) element.getConfig()).getCollectionsLabel();
+      final var value = collectionsLabel + "\n" + icfCodes + "\n" + "\n" + text;
+
+      return CertificateQuestionValueText.builder()
+          .value(value)
+          .build();
+    }
+    return getText(text);
```

Comment:
What do you think about breaking out the if-statement above into a method?

By doing that, you could flatten out this method and just return something like this:
```java
return value.getIcfCodes != null && !value.getIcfCodes.isEmpty() ? getTextWithIcfCodes() : getText();
```